### ISSUE TITLE
[2020 INTERSPEECH] In defence of metric learning for speaker recognition

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#99]
기존의 통념은 speaker recognition task에서 classification objectives가 metric learning objectives보다 outperform한다는 것이었음

저자들은 `open-set` speaker recognition task에서 metric learning objectives가 성능이 더 좋으며, `Angular Prototypical Loss`를 제안함

기존의 loss들은 간단하게만 설명

## 1. Training Functions
<img width="800" alt="image" src="https://user-images.githubusercontent.com/49643709/202049313-7c8aace8-56e2-4a7d-b8da-9233f96efc08.png">

Task의 목적은 representation이 low intra-class variance, high inter-class variance를 가지도록 하는 것임

#### 1.1.1. Softmax
<img width="400" alt="image" src="https://user-images.githubusercontent.com/49643709/202047904-52d36b12-6ff4-4aa2-abbb-f05664548f38.png">

Classification에서 흔히 사용하는 loss

해당 loss는 classification error에만 penalty를 주기에, intra-class compactness와 inter-class separation을 explicitly enforce할 수 없음

#### 1.1.2. AM-Softmax (CosFace)
<img width="400" alt="image" src="https://user-images.githubusercontent.com/49643709/202047987-696583b3-1039-4b45-9625-426b2862dbab.png">

Normalised Softmax Loss (NSL)에 cosine margin m, fixed scale factor s를 붙인 loss

Cosine margin을 붙인 이유는 NSL만 사용하면 classification error만 penalise하기에 discriminative하다고 보기 sufficient하지 않아서임

Fixed scale factor을 붙인 이유는 training phase에서 gradient가 너무 작아지는 것을 방지하기 위해서임

#### 1.1.3. AAM-Softmax (ArcFace)
<img width="400" alt="image" src="https://user-images.githubusercontent.com/49643709/202048018-c90a228b-6ba6-4be1-a3fa-e4c36f21a04f.png">

CosFace에 additive angular margin penalty m을 붙인 loss

#### 1.2.1. Triplet
<img width="400" alt="image" src="https://user-images.githubusercontent.com/49643709/202050929-2560d03c-fb07-4418-a5ca-86b3a3ff4705.png">

Query data에 대해 1개의 positive에 대해서는 L2 distance가 작아지게, 1개의 negative에 대해서는 L2 distance가 커지게하는 loss

#### 1.2.2. Prototypical
<img width="400" alt="image" src="https://user-images.githubusercontent.com/49643709/202051235-22036a3a-0eee-41f4-9046-82acee6b0c61.png">

Query data에 대해 support set으로 class별 centroid를 구하고, Euclidean distance를 계산하여 같은 class는 가까워지게, 다른 class는 멀어지게하는 loss

<img width="800" alt="image" src="https://user-images.githubusercontent.com/49643709/202055568-97f60ed8-9f84-42ce-9b7d-99afd058883c.png">

Implementation을 보면 편의상 batch 내에서 speaker별 0번째 utterances들을 anchor로, 나머지 utterances를 이용하여 centroid를 구함

#### 1.2.3. Generalised end-to-end (GE2E)
<img width="400" alt="image" src="https://user-images.githubusercontent.com/49643709/202051329-384eb7db-c82c-428a-8d9e-bd227d650157.png">

Batch 내의 모든 utterance에 대하여 batch내의 utterances들로 class별 centroid를 구하고, cosine distance를 계산하여 같은 class는 가까워지게, 다른 class는 멀어지게하는 loss

#### 1.2.4. Angular Prototypical
<img width="400" alt="image" src="https://user-images.githubusercontent.com/49643709/202054977-218478c1-2d9b-41bf-83d6-a5be6e17ead9.png">

저자들이 제안한 loss로, Prototypical loss와 똑같은데 distance만 scaled cosine distance로 바꿈

## 2. Experiments
<img width="800" alt="image" src="https://user-images.githubusercontent.com/49643709/202047764-d26fb08c-1c69-4f2d-97ee-571f5378b0e5.png">
<img width="800" alt="image" src="https://user-images.githubusercontent.com/49643709/202047799-90d4ce8e-511f-4ed1-aa19-0a7a639732ba.png">

Metric learning objectives가 classification objectives보다 성능이 좋음

### 2.1. Implementation details

AAM-Softmax loss, triplet loss가 학습하는데 있어 diverge하는 현상이 있어 curriculum learning을 적용

AAM-Softmax loss의 경우 hyperparameter m을 0.1에서 100 epoch 학습한 이후 0.3으로 증가시킴

Triplet loss의 경우 100 epoch 학습한 이후 hard negative mining을 적용

### 2.2. Evaluation results

`Table 3`를 보면 알 수 있듯이, metric learning methods의 경우 large batch size를 적용하면 성능이 오름

성능이 오르는 이유는 batch size가 커질수록 harder negatives가 batch 내에 있을 확률이 더 높아지기 때문

Classification loss에서는 batch size가 커진다고 성능에 영향을 주지 않음
